### PR TITLE
[DONOTMERGE] Refactor cli factory module

### LINKED ISSUE #2353: Automation failure : test_contenthost_apply_errata
TestNames:
tests.foreman.cli.test_contenthost.TestCHKatelloAgent.test_contenthost_apply_errata

robottelo.common.ssh: DEBUG: >>> [qe-foreman-rhel66.usersys.redhat.com] LANG=en_US.UTF-8 hammer -v -u admin -p changeme  content-view version promote --id="219" --organization-id="219" --to-lifecycle-environment-id="225"
robottelo.common.ssh: INFO: Instantiated Paramiko client 0x433ecd0
robottelo.common.ssh: INFO: Destroying Paramiko client 0x433ecd0
robottelo.common.ssh: INFO: Destroyed Paramiko client 0x433ecd0
robottelo.common.ssh: DEBUG: <<< stderr
[ERROR 2015-05-31 21:10:06 Exception] Required lock is already taken by other running tasks.
Please inspect their state, fix their errors and resume them.

Required lock: read
Conflicts with tasks:
- https://qe-foreman-rhel66.usersys.redhat.com/foreman_tasks/tasks/360043ca-8b28-4078-9f2f-b5c43beb7f44
Could not promote the content view:
  Required lock is already taken by other running tasks.
  Please inspect their state, fix their errors and resume them.
  
  Required lock: read
  Conflicts with tasks:
  - https://qe-foreman-rhel66.usersys.redhat.com/foreman_tasks/tasks/360043ca-8b28-4078-9f2f-b5c43beb7f44
[ERROR 2015-05-31 21:10:06 Exception] 

RestClient::InternalServerError (500 Internal Server Error):
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/abstract_response.rb:48:in `return!'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/request.rb:230:in `process_result'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/request.rb:178:in `transmit'
    /usr/lib/ruby/1.8/net/http.rb:543:in `start'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/request.rb:172:in `transmit'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/request.rb:64:in `execute'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/request.rb:33:in `execute'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/resource.rb:71:in `post'
    /usr/lib/ruby/gems/1.8/gems/apipie-bindings-0.0.13/lib/apipie_bindings/api.rb:280:in `send'
    /usr/lib/ruby/gems/1.8/gems/apipie-bindings-0.0.13/lib/apipie_bindings/api.rb:280:in `call_client'
    /usr/lib/ruby/gems/1.8/gems/apipie-bindings-0.0.13/lib/apipie_bindings/api.rb:211:in `http_call'
    /usr/lib/ruby/gems/1.8/gems/apipie-bindings-0.0.13/lib/apipie_bindings/api.rb:162:in `call'
    /usr/lib/ruby/gems/1.8/gems/apipie-bindings-0.0.13/lib/apipie_bindings/resource.rb:14:in `call'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/lib/hammer_cli/./apipie/command.rb:43:in `send_request'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli_foreman-0.2.0/lib/hammer_cli_foreman/commands.rb:185:in `send_request'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli_foreman_tasks-0.0.6/lib/hammer_cli_foreman_tasks/async_command.rb:14:in `execute'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/command.rb:68:in `run'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/lib/hammer_cli/./apipie/../abstract.rb:23:in `run'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/subcommand/execution.rb:11:in `execute'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/command.rb:68:in `run'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/lib/hammer_cli/./apipie/../abstract.rb:23:in `run'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/subcommand/execution.rb:11:in `execute'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/command.rb:68:in `run'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/lib/hammer_cli/./apipie/../abstract.rb:23:in `run'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/subcommand/execution.rb:11:in `execute'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/command.rb:68:in `run'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/lib/hammer_cli/./apipie/../abstract.rb:23:in `run'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/command.rb:126:in `run'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/bin/hammer:108
    /usr/bin/hammer:19:in `load'
    /usr/bin/hammer:19

Stacktrace

  File "/usr/lib64/python2.7/unittest/case.py", line 360, in run
    self.setUp()
  File "/home/jenkins/workspace/satellite6-automation-upstream-rhel66-cli/tests/foreman/cli/test_contenthost.py", line 594, in setUp
    u'activationkey-id': TestCHKatelloAgent.activation_key['id'],
  File "robottelo/common/decorators.py", line 76, in cacheable_function
    new_object = func(options)
  File "robottelo/cli/factory.py", line 1801, in setup_org_for_a_rh_repo
    raise CLIFactoryError('Failed to promote version to next environment')
'Failed to promote version to next environment\n

### LINKED ISSUE #2354: Automation failure : test_contenthost_get_errata_info
TestNames:
tests.foreman.cli.test_contenthost.TestCHKatelloAgent.test_contenthost_get_errata_info

robottelo.common.ssh: DEBUG: >>> [qe-foreman-rhel66.usersys.redhat.com] LANG=en_US.UTF-8 hammer -v -u admin -p changeme --output=csv repository-set enable --product="Red Hat Enterprise Linux Server" --basearch="x86_64" --releasever="7Server" --name="Red Hat Satellite Tools 6 Beta (for RHEL 7 Server) (RPMs)" --organization-id="219"
robottelo.common.ssh: INFO: Instantiated Paramiko client 0x4418d50
robottelo.common.ssh: INFO: Destroying Paramiko client 0x4418d50
robottelo.common.ssh: INFO: Destroyed Paramiko client 0x4418d50
robottelo.common.ssh: DEBUG: <<< stderr
[ERROR 2015-05-31 21:10:26 Exception] Error: 409 Conflict
Could not enable repository:
  Error: 409 Conflict
[ERROR 2015-05-31 21:10:26 Exception] 

RestClient::Conflict (409 Conflict):
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/abstract_response.rb:48:in `return!'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/request.rb:230:in `process_result'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/request.rb:178:in `transmit'
    /usr/lib/ruby/1.8/net/http.rb:543:in `start'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/request.rb:172:in `transmit'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/request.rb:64:in `execute'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/request.rb:33:in `execute'
    /usr/lib/ruby/gems/1.8/gems/rest-client-1.6.7/lib/restclient/resource.rb:80:in `put'
    /usr/lib/ruby/gems/1.8/gems/apipie-bindings-0.0.13/lib/apipie_bindings/api.rb:280:in `send'
    /usr/lib/ruby/gems/1.8/gems/apipie-bindings-0.0.13/lib/apipie_bindings/api.rb:280:in `call_client'
    /usr/lib/ruby/gems/1.8/gems/apipie-bindings-0.0.13/lib/apipie_bindings/api.rb:211:in `http_call'
    /usr/lib/ruby/gems/1.8/gems/apipie-bindings-0.0.13/lib/apipie_bindings/api.rb:162:in `call'
    /usr/lib/ruby/gems/1.8/gems/apipie-bindings-0.0.13/lib/apipie_bindings/resource.rb:14:in `call'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/lib/hammer_cli/./apipie/command.rb:43:in `send_request'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli_foreman-0.2.0/lib/hammer_cli_foreman/commands.rb:185:in `send_request'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/lib/hammer_cli/./apipie/command.rb:34:in `execute'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/command.rb:68:in `run'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/lib/hammer_cli/./apipie/../abstract.rb:23:in `run'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/subcommand/execution.rb:11:in `execute'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/command.rb:68:in `run'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/lib/hammer_cli/./apipie/../abstract.rb:23:in `run'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/subcommand/execution.rb:11:in `execute'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/command.rb:68:in `run'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/lib/hammer_cli/./apipie/../abstract.rb:23:in `run'
    /usr/lib/ruby/gems/1.8/gems/clamp-0.6.2/lib/clamp/command.rb:126:in `run'
    /usr/lib/ruby/gems/1.8/gems/hammer_cli-0.2.0/bin/hammer:108
    /usr/bin/hammer:19:in `load'
    /usr/bin/hammer:19

Stacktrace

  File "/usr/lib64/python2.7/unittest/case.py", line 360, in run
    self.setUp()
  File "/home/jenkins/workspace/satellite6-automation-upstream-rhel66-cli/tests/foreman/cli/test_contenthost.py", line 594, in setUp
    u'activationkey-id': TestCHKatelloAgent.activation_key['id'],
  File "robottelo/common/decorators.py", line 76, in cacheable_function
    new_object = func(options)
  File "robottelo/cli/factory.py", line 1759, in setup_org_for_a_rh_repo
    raise CLIFactoryError('Failed to enable repository set')
'Failed to enable repository set\n--------------------

### PR DESCRIPTION
**Pull request by elyezer**
The dict method get already returns None if the key is not present.
Drop all None values for the second argument on dict.get.

    In [1]: d = {}

    In [2]: print d.get('key')
    None

Also get the releasever from options on setup_org_for_a_rh_repo
function. This will avoid passing the releasever option when enabling
Satellite 6 Tools repo.

Check for every result return code in setup_org_for_a_rh_repo in order
to fail earlier.

Closes #2354 and closes #2353.

### CODE REVIEW

File: robottelo/cli/factory.py [hunk 1]
```diff
@@ -1820,26 +1820,28 @@ def setup_org_for_a_rh_repo(options=None):
         raise CLIFactoryError('Failed to upload manifest')
     # Enable repo from Repository Set
     result = RepositorySet.enable({
-        u'name': options.get('repository-set', None),
+        u'name': options.get('repository-set'),
         u'organization-id': org_id,
-        u'product': options.get('product', None),
-        u'releasever': '7Server',
+        u'product': options.get('product'),
+        u'releasever': options.get('releasever'),
```
Comment (elyezer) [inline robottelo/cli/factory.py]: If this is not specified None will be returned and the option will not be passed to the built hammer command.

**Comment by elyezer** [comment]
This depends on #2633.

**Comment by abalakh** [comment]
@elyezer please take a look at #2649 . For `setup_org_for_a_rh_repo` some of your changes are already there, and some are conflicting (e.g., few `dict.get()` calls were removed and replaced with `if ... else` statements)

File: robottelo/cli/factory.py (hunk 1 above)
Comment (abalakh) [inline robottelo/cli/factory.py]: Nice change, but please update docstring with new optional parameter then :)

File: robottelo/cli/factory.py [hunk 2]
```diff
@@ -1861,6 +1863,8 @@ def setup_org_for_a_rh_repo(options=None):
         raise CLIFactoryError('Failed to publish new version of content view')
     # Get the version id
     result = ContentView.info({u'id': cv_id})
+    if result.return_code != 0:
+        raise CLIFactoryError('Failed to fecth content view info')
```
Comment (Ichimonji10) [inline robottelo/cli/factory.py]: "fecth"

**Comment by Ichimonji10** [comment]
ACK, but beware [merge conflicts](#issuecomment-128368688) as mentioned by @abalakh.

File: robottelo/cli/factory.py (hunk 2 above)
Comment (elyezer) [inline robottelo/cli/factory.py]: thanks

File: robottelo/cli/factory.py (hunk 1 above)
Comment (elyezer) [inline robottelo/cli/factory.py]: :+1:

**Comment by sthirugn** [comment]
Lets wait and merge this after https://github.com/SatelliteQE/robottelo/pull/2649 is merged?

**Comment by elyezer** [comment]
Yes, please merge this only after other two PRs getting merged. I will adjust the conflicts before. Marking as do not merge for while.

**Comment by elyezer** [comment]
Closing this in favor of #2659

**Pull request closed without merging**